staking-pool/src/lib.rs

```diff
@@ -378,12 +385,19 @@ impl StakingContract {
 
     /// Restakes the current `total_staked_balance` again.
     fn restake(&mut self) {
+        if self.paused {
+            return;
```

Comment:
should we return something that indicates the contract has been paused? Otherwise if people call `stake`, they will think that they have successfully staked and may be confused if they find out that they didn't actually stake.

Reply:
Well, they are staked successfully. And the pool will return to normal, once the staking is resumed. People can verify the performance of the pool in the past before staking.